components/idp-mgt/org.wso2.carbon.idp.mgt/src/main/java/org/wso2/carbon/idp/mgt/IdentityProviderManager.java

```diff
@@ -169,9 +188,24 @@ public IdentityProvider getResidentIdP(String tenantDomain)
             openIdUrl = IdentityUtil.getServerURL(IdentityConstants.OpenId.OPENID, true, true);
         }
 
-        samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true, true, false);
+        if (IdentityTenantUtil.isTenantQualifiedUrlsEnabled()) {
+
+            try {
+                samlSSOUrl = ServiceURLBuilder.create().addPath(IdentityConstants.ServerConfig.SAMLSSO).build()
+                        .getAbsoluteURL();
+                samlLogoutUrl = ServiceURLBuilder.create().addPath(IdentityConstants.ServerConfig.SAMLSSO).build()
+                        .getAbsoluteURL();
+            } catch (URLBuilderException e) {
+                throw IdentityRuntimeException
+                        .error("Error while building URL: " + IdentityConstants.ServerConfig.SAMLSSO, e);
+            }
+        } else {
+            samlSSOUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true)
+                    + IdPManagementUtil.getTenantParameter();
 
-        samlLogoutUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true, true, false);
+            samlLogoutUrl = IdentityUtil.getServerURL(IdentityConstants.ServerConfig.SAMLSSO, true, true)
+                    + IdPManagementUtil.getTenantParameter();
```

Comment:
were these not read from the file before

Reply:
I reverted to previous changes [1]. There, samlLogoutUrl and samlSSOUrl had been using the IdentityConstants.

[1] - []()https://github.com/wso2/carbon-identity-framework/pull/2828/files#diff-98664c0777759c45d183f9180c27614e